libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt

```diff
@@ -85,7 +91,7 @@ class NavigationController {
                 )
             )
             OffboardRouter -> arrayOf(
-                Context::class.java to context
+                NavigationOffboardRoute::class.java to navigationOffboardRoute
```

Comment:
This leaks `NavigationOffboardRoute` to the controller. Why can't we use the `(accessToken, context)` constructor?

Reply:
`MapboxDirection` checks if access token is valid by pattern, so if I create it under the hood it throws exception

Reply:
I believe that @LukasPaczos is referring to implement something like this 👇 

```diff
diff --git a/libandroid-navigation-ui/src/main/assets/sdk_versions/com.mapbox.services.android.navigation.ui.v5 b/libandroid-navigation-ui/src/main/assets/sdk_versions/com.mapbox.services.android.navigation.ui.v5
deleted file mode 100644
index 5cf16b8c..00000000
--- a/libandroid-navigation-ui/src/main/assets/sdk_versions/com.mapbox.services.android.navigation.ui.v5
+++ /dev/null
@@ -1,2 +0,0 @@
-libandroid-navigation-ui/0.43.0-SNAPSHOT
-v-1
diff --git a/libandroid-navigation/src/main/assets/sdk_versions/com.mapbox.services.android.navigation b/libandroid-navigation/src/main/assets/sdk_versions/com.mapbox.services.android.navigation
deleted file mode 100644
index 9af9ac52..00000000
--- a/libandroid-navigation/src/main/assets/sdk_versions/com.mapbox.services.android.navigation
+++ /dev/null
@@ -1,2 +0,0 @@
-libandroid-navigation/0.43.0-SNAPSHOT
-v-1
diff --git a/libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt b/libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt
index d59e71d2..dddc0267 100644
--- a/libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt
+++ b/libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt
@@ -21,7 +21,6 @@ import com.mapbox.navigation.base.route.model.Route
 import com.mapbox.navigation.directions.session.DirectionsSession
 import com.mapbox.navigation.module.NavigationModuleProvider
 import com.mapbox.navigation.navigator.MapboxNativeNavigator
-import com.mapbox.navigation.route.offboard.router.NavigationOffboardRoute
 import com.mapbox.navigation.trip.notification.NavigationNotificationProvider
 import com.mapbox.navigation.trip.service.TripService
 import com.mapbox.navigation.trip.session.TripSession
@@ -35,9 +34,9 @@ class NavigationController {
     private val navigator: MapboxNativeNavigator
     private val locationEngine: LocationEngine
     private val locationEngineRequest: LocationEngineRequest
-    private val navigationOffboardRoute: NavigationOffboardRoute
     private val navigationNotificationProvider: NavigationNotificationProvider
     private val context: Context
+    private val accessToken: String
 
     private val mainHandler: Handler by lazy { Handler(Looper.getMainLooper()) }
     private val workerHandler: Handler by lazy { Handler(workerThread.looper) }
@@ -52,19 +51,19 @@ class NavigationController {
 
     constructor(
         context: Context,
+        accessToken: String,
         navigator: MapboxNativeNavigator,
         locationEngine: LocationEngine,
         locationEngineRequest: LocationEngineRequest,
         tripServiceLambda: () -> Unit,
-        navigationOffboardRoute: NavigationOffboardRoute,
         navigationNotificationProvider: NavigationNotificationProvider
     ) {
         this.context = context
+        this.accessToken = accessToken
         this.navigator = navigator
         this.locationEngine = locationEngine
         this.locationEngineRequest = locationEngineRequest
         this.navigationNotificationProvider = navigationNotificationProvider
-        this.navigationOffboardRoute = navigationOffboardRoute
 
         logger = NavigationModuleProvider.createModule(LoggerModule, ::paramsProvider)
         directionsSession = NavigationComponentProvider.createDirectionsSession(
@@ -117,7 +116,8 @@ class NavigationController {
                 )
             )
             OffboardRouter -> arrayOf(
-                NavigationOffboardRoute::class.java to navigationOffboardRoute
+                String::class.java to accessToken,
+                Context::class.java to context
             )
             OnboardRouter -> arrayOf(
                 MapboxNativeNavigator::class.java to navigator
diff --git a/libnavigation-core/src/test/java/com/mapbox/navigation/NavigationControllerTest.kt b/libnavigation-core/src/test/java/com/mapbox/navigation/NavigationControllerTest.kt
index 1d6ad48f..6aac78db 100644
--- a/libnavigation-core/src/test/java/com/mapbox/navigation/NavigationControllerTest.kt
+++ b/libnavigation-core/src/test/java/com/mapbox/navigation/NavigationControllerTest.kt
@@ -5,7 +5,6 @@ import android.content.Context
 import com.mapbox.android.core.location.LocationEngine
 import com.mapbox.android.core.location.LocationEngineRequest
 import com.mapbox.navigation.navigator.MapboxNativeNavigator
-import com.mapbox.navigation.route.offboard.router.NavigationOffboardRoute
 import com.mapbox.navigation.trip.notification.NavigationNotificationProvider
 import com.mapbox.navigation.utils.extensions.inferDeviceLocale
 import io.mockk.every
@@ -25,10 +24,10 @@ class NavigationControllerTest {
 
     private lateinit var navigationController: NavigationController
     private val context: Context = mockk(relaxed = true)
+    private val accessToken: String = "pk.XXX"
     private val navigator: MapboxNativeNavigator = mockk()
     private val locationEngine: LocationEngine = mockk()
     private val locationEngineRequest: LocationEngineRequest = mockk()
-    private val navigationOffboardRoute: NavigationOffboardRoute = mockk()
     private val navigationNotificationProvider: NavigationNotificationProvider = mockk()
     private val tripServiceLambda: () -> Unit = mockk()
 
@@ -49,11 +48,11 @@ class NavigationControllerTest {
         navigationController =
             NavigationController(
                 context,
+                accessToken,
                 navigator,
                 locationEngine,
                 locationEngineRequest,
                 tripServiceLambda,
-                navigationOffboardRoute,
                 navigationNotificationProvider
             )
     }
```

Reply:
> `MapboxDirection` checks if access token is valid by pattern, so if I create it under the hood it throws exception 

Yeah, that's correct 👀 

https://github.com/mapbox/mapbox-java/blob/3d4297ec822275edb41849c95bfef5f6c4c1615b/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L966-L969

https://github.com/mapbox/mapbox-java/blob/3d4297ec822275edb41849c95bfef5f6c4c1615b/services-core/src/main/java/com/mapbox/core/utils/MapboxUtils.java#L23-L26

Note

```kotlin
private val accessToken: String = "pk.XXX"
```

Using a "valid" access token (`"pk."`) works 👌

Reply:
@Guardiola31337 for this I need valid access token, because as I said `DirectionSession` checks one under the hood